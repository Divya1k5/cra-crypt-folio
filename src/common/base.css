html * {
    box-sizing: border-box;
}


body {
    margin: 0;
    padding: 0;
    font-family: -apple-system, BlinkMacSystemFont, "Helvetica Neue";
}

.module {
    padding: var(--external-padding);
}

div {
    line-height: 1.2em;
}