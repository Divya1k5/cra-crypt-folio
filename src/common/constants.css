html {
    --primary-color: #0277bd;
    --secondary-color: #63dcfc;
    --external-padding: 20px;
    --internal-padding: 15px;
    --internal-padding-small: 10px;
    --internal-padding-smaller: 5px;
    --input-field-height: 20px;
    --transition-time-slow: 0.7s;
    --transition-time-fast: 0.5s;
    --font-size: 14px;
    --card-size: 250px;
    --card-border-radius: 10px;
    --border-thickness: 2px;
    --notification-bar-height: 50px;
}