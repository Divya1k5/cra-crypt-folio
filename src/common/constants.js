export const api = {
    'BTC': 'https://api.coinbase.com/v2/prices/BTC-USD/buy',
    'ETH': 'https://api.coinbase.com/v2/prices/ETH-USD/buy',
    'LTC': 'https://api.coinbase.com/v2/prices/LTC-USD/buy'
};

export const coinTypes = ['BTC','ETH','LTC'];

export const localStorageKey = 'cryptfolio';