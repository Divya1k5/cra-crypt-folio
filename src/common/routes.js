import React from 'react'
import { Switch, Route } from 'react-router';
import HomePage from '../pages/HomePage';
import SummaryPage from '../pages/SummaryPage';
import NotFoundPage from '../pages/NotFoundPage';

export const routes = (
    <Switch>
        <Route exact path="/" component={HomePage} />
        <Route exact path="/summary" component={SummaryPage} />
        <Route component={NotFoundPage} />
    </Switch>
);
