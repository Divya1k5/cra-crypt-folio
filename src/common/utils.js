export const setObjInLocalStorage = (key, obj) => {
    localStorage.setItem(key, JSON.stringify(obj));
};

export const getObjFromLocalStorage = (key) => {
    return JSON.parse(localStorage.getItem(key));
};

export const isMobileDevice = () => {
    if( /Android|webOS|iPhone|iPad|iPod|Opera Mini/i.test(navigator.userAgent) ) {
        return true;
    }
};