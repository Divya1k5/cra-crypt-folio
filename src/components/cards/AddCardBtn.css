.addCardBtnMenu {
    position: fixed;
    bottom: 50px;
    right: 50px;
}

@media (max-width: 400px) {
    .addCardBtnMenu {
        bottom: 100px;
    }
}

.addCardBtnMenu .primaryBtn {
    font-size: 35px;
    padding-top: 5px;
}

.addCardBtnMenu .target {
    border-radius: 50%;
    height: 60px;
    width: 60px;
    background-color: var(--primary-color);
    color: var(--secondary-color);
}

.addCardBtnMenu .target:hover {
    cursor: pointer;
}

.addCardBtnMenu .list {
    padding: 0;
}

.addCardBtnMenu .list .item {
    display: flex;
    align-items: center;
    justify-content: center;
    position: absolute;
    left: 0;
}