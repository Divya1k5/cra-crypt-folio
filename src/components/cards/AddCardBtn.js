import React, { Component } from 'react';
import './AddCardBtn.css';
import MdAdd from 'react-icons/lib/md/add';
import MdClose from 'react-icons/lib/md/close';
import { coinTypes } from '../../common/constants';

export default class AddCardBtn extends Component {

    constructor(props) {
        super(props);

        this.state = {
            isMenuOpen: false
        }
    }

    onAddBtnClick() {
        this.setState({
            isMenuOpen: !this.state.isMenuOpen
        });

    }

    addCard(type) {
        this.setState({
            isMenuOpen: !this.state.isMenuOpen
        });
        if (typeof this.props.addCard === 'function') {
            this.props.addCard(type);
        }
    }

    render() {

        const menuItemSeparatorHeight = 80;

        return (
            <div className="addCardBtnMenu">

                <div className="primaryBtn target" onClick={this.onAddBtnClick.bind(this)}>
                    {this.state.isMenuOpen ? (
                        <div className="cta"><MdClose /></div>
                    ) : (
                        <div className="cta"><MdAdd /></div>
                    )}
                </div>

                <div className="list">

                    {this.state.isMenuOpen ?
                        
                        coinTypes.map((coinType, idx) => {

                            const itemStyle = {
                                top: (-1 * ((idx + 1) * menuItemSeparatorHeight)) + 'px'
                            };
                            return <div key={coinType} className="item cta target" style={itemStyle} onClick={this.addCard.bind(this, coinType)}>{coinType}</div>;
                        })
                        : null
                    }

                </div>

            </div>
        );
    }
}