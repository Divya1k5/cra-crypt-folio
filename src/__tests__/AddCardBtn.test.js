import React from 'react';
import AddCardBtn from '../components/cards/AddCardBtn';
import renderer from 'react-test-renderer';
import { coinTypes } from '../common/constants';
import { shallow } from "enzyme";

describe('Shallow DOM Testing on AddCardBtn', () => {
    let menu;

    beforeEach(() => {
        menu = shallow(<AddCardBtn />);
    });

    it('AddCardBtn opens menu after one click', ()=> {
        menu.find('.primaryBtn').simulate('click');
        expect(menu.find('.item .cta').length === coinTypes.length);
    });

    it('AddCardBtn opens menu with correct coin names', ()=> {
        menu.find('.primaryBtn').simulate('click');
        menu.find('.item .cta').map((item, index) => {
            expect(item.text()).toEqual(coinTypes[index]);
        });
    });

    it('AddCardBtn closes menu after two clicks', ()=> {
        menu.find('.primaryBtn').simulate('click');
        menu.find('.primaryBtn').simulate('click');
        expect(menu.find('.item .cta').length === 0);
    });
});

describe('Snapshot Testing on AddCardBtn', () => {

    it('AddCardBtn renders correctly', () => {
        const tree = renderer.create(<AddCardBtn />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('AddCardBtn renders opens and closes menu', () => {

        const component = renderer.create(<AddCardBtn />);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        const primaryBtn = tree.children.find((child) => child.props["data-test"] === 'primaryBtn');

        //manually trigger a click
        primaryBtn.props.onClick();

        //re-rendering the tree, menu should open
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        //manually trigger a click
        primaryBtn.props.onClick();

        //re-rendering the tree, menu should close
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

    });
});

