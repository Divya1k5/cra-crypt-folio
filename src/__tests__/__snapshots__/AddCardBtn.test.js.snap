// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing on AddCardBtn AddCardBtn renders correctly 1`] = `
<div
  className="addCardBtnMenu"
>
  <div
    className="primaryBtn target"
    data-test="primaryBtn"
    onClick={[Function]}
  >
    <div
      className="cta"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m31.6 21.6h-10v10h-3.2v-10h-10v-3.2h10v-10h3.2v10h10v3.2z"
          />
        </g>
      </svg>
    </div>
  </div>
  <div
    className="list"
  />
</div>
`;

exports[`Snapshot Testing on AddCardBtn AddCardBtn renders opens and closes menu 1`] = `
<div
  className="addCardBtnMenu"
>
  <div
    className="primaryBtn target"
    data-test="primaryBtn"
    onClick={[Function]}
  >
    <div
      className="cta"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m31.6 21.6h-10v10h-3.2v-10h-10v-3.2h10v-10h3.2v10h10v3.2z"
          />
        </g>
      </svg>
    </div>
  </div>
  <div
    className="list"
  />
</div>
`;

exports[`Snapshot Testing on AddCardBtn AddCardBtn renders opens and closes menu 2`] = `
<div
  className="addCardBtnMenu"
>
  <div
    className="primaryBtn target"
    data-test="primaryBtn"
    onClick={[Function]}
  >
    <div
      className="cta"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m31.6 10.7l-9.3 9.3 9.3 9.3-2.3 2.3-9.3-9.3-9.3 9.3-2.3-2.3 9.3-9.3-9.3-9.3 2.3-2.3 9.3 9.3 9.3-9.3z"
          />
        </g>
      </svg>
    </div>
  </div>
  <div
    className="list"
  >
    <div
      className="item cta target"
      onClick={[Function]}
      style={
        Object {
          "top": "-80px",
        }
      }
    >
      BTC
    </div>
    <div
      className="item cta target"
      onClick={[Function]}
      style={
        Object {
          "top": "-160px",
        }
      }
    >
      ETH
    </div>
    <div
      className="item cta target"
      onClick={[Function]}
      style={
        Object {
          "top": "-240px",
        }
      }
    >
      LTC
    </div>
  </div>
</div>
`;

exports[`Snapshot Testing on AddCardBtn AddCardBtn renders opens and closes menu 3`] = `
<div
  className="addCardBtnMenu"
>
  <div
    className="primaryBtn target"
    data-test="primaryBtn"
    onClick={[Function]}
  >
    <div
      className="cta"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m31.6 21.6h-10v10h-3.2v-10h-10v-3.2h10v-10h3.2v10h10v3.2z"
          />
        </g>
      </svg>
    </div>
  </div>
  <div
    className="list"
  />
</div>
`;
